OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitOnlyOnYoutubeFragment.java

```diff
@@ -15,34 +14,59 @@
 
 public class CourseUnitOnlyOnYoutubeFragment extends CourseUnitFragment {
 
-    public static CourseUnitOnlyOnYoutubeFragment newInstance(CourseComponent unit) {
+    private String youtubeApiKey;
+    private YouTubeMediaPlayerSupportFragment youTubePlayerFragment;
+    private boolean startVideoWhileInitializingYoutubeVideo;
+
+    public static CourseUnitOnlyOnYoutubeFragment newInstance(CourseComponent unit, String youtubeApiKey) {
         CourseUnitOnlyOnYoutubeFragment fragment = new CourseUnitOnlyOnYoutubeFragment();
         Bundle args = new Bundle();
         args.putSerializable(Router.EXTRA_COURSE_UNIT, unit);
+        args.putString(Router.EXTRA_YOUTUBE_API_KEY, youtubeApiKey);
         fragment.setArguments(args);
-
         return fragment;
     }
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        youtubeApiKey = getArguments() == null ? null :
+                getArguments().getString(Router.EXTRA_YOUTUBE_API_KEY);
     }
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
-        View v = inflater.inflate(R.layout.fragment_course_unit_only_on_youtube, container, false);
-        ((TextView) v.findViewById(R.id.only_youtube_available_message)).setText(R.string.assessment_only_on_youtube);
-        v.findViewById(R.id.view_on_youtube_button).setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                Intent i = new Intent(Intent.ACTION_VIEW);
-                i.setData(Uri.parse(((VideoBlockModel) unit).getData().encodedVideos.youtube.url));
-                startActivity(i);
-            }
-        });
-        return v;
+        return inflater.inflate(R.layout.fragment_course_unit_only_on_youtube, container, false);
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        // If youtube fragment is first page in viewPager, being stated that "onPageShow" isn't called.
+        // We have to handle the initial play for first youtube fragment ourselves
+        if (startVideoWhileInitializingYoutubeVideo)
```

Comment:
use `initializePlayBack` instead of ` startVideoWhileInitializingYoutubeVideo`.

Reply:
'shouldInitializePlayback` makes more sense in this scenario.